pkg/controllers/status/cluster_status_controller.go

```diff
@@ -195,6 +201,30 @@ func (c *ClusterStatusController) syncClusterStatus(cluster *clusterv1alpha1.Clu
 	return c.updateStatusIfNeeded(cluster, currentClusterStatus)
 }
 
+func (c *ClusterStatusController) retryOnFailure(online, healthy bool, cluster string) bool {
```

Comment:
I can't tell but seems something doesn't feel right here.
Let me think about it.

Reply:
Have modified as you suggested